benchmark/networkx_benchmark.py

```diff
@@ -27,36 +49,41 @@ def _run_algorithm(self):
             self._logger.info(self._logger_prefix + f"Succesfully ran community detection.")
+            write_com_to_file(self._communities, get_com_path(self._config))
 
         elif (self._config.algorithm == "async_fluid"):
-            self._communities = self._measure_time_and_get_results(community.asyn_fluidc, self._graph, self._k)
-            self._logger.info(self._logger_prefix + f"Succesfully ran community detection.")
-
-        elif (self._config.algorithm == "asyn_lpa_communities"):
-            self._communities = self._measure_time_and_get_results(community.asyn_lpa_communities, self._graph)
+            self._communities = self._measure_time_and_get_results(async_fluid, self._graph, self._expected_communities_count)
             self._logger.info(self._logger_prefix + f"Succesfully ran community detection.")
+            write_com_to_file(self._communities, get_com_path(self._config))
 
         elif (self._config.algorithm == "asyn_lpa_communities"):
-            self._communities = self._measure_time_and_get_results(community.asyn_lpa_communities, self._graph)
+            self._communities = self._measure_time_and_get_results(async_lpa_communities, self._graph)
             self._logger.info(self._logger_prefix + f"Succesfully ran community detection.")
+            write_com_to_file(self._communities, get_com_path(self._config))
 
         elif (self._config.algorithm == "label_propagation_communities"):
-            self._communities = self._measure_time_and_get_results(community.label_propagation_communities, self._graph)
+            self._communities = self._measure_time_and_get_results(label_propagation_communities, self._graph)
             self._logger.info(self._logger_prefix + f"Succesfully ran community detection.")
+            write_com_to_file(self._communities, get_com_path(self._config))
 
         elif (self._config.algorithm == "lukes_partitioning"):
             self._communities = self._measure_time_and_get_results(community.lukes_partitioning, self._graph, self._max_weight)
             self._logger.info(self._logger_prefix + f"Succesfully ran community detection.")
+            write_com_to_file(self._communities, get_com_path(self._config))
 
         elif (self._config.algorithm == "greedy_modularity_communities"):
             self._communities = self._measure_time_and_get_results(community.greedy_modularity_communities, self._graph)
             self._logger.info(self._logger_prefix + f"Succesfully ran community detection.")
+            write_com_to_file(self._communities, get_com_path(self._config))
 
         elif (self._config.algorithm == "k_clique_communities"):
             self._communities = self._measure_time_and_get_results(community.k_clique_communities, self._graph, self._k)
             self._logger.info(self._logger_prefix + f"Succesfully ran community detection.")
+            write_com_to_file(self._communities, get_com_path(self._config))
```

Comment:
Also es ist wahrscheinlich nicht Sinn und Zweck alle Results immer zu speichern oder? Um jetzt kein kompletten Caching-Schnack zu bauen, hätte ich vorgeschlagen das man noch eine Flag in der config hinzufügt, so dass Results nur gespeichert werden wenn man da `true`setzt .... die Results von allen deterministischen Algos sollten sich ja ohnehin nicht enden ...

Reply:
Ist in den changes drin